libdevcrypto/OverlayDB.cpp

```diff
@@ -155,6 +155,20 @@ void OverlayDB::kill(h256 const& _h)
 #endif
 }
 
+bool OverlayDB::deepkill(h256 const& _h)
+{
+	// kill in memoryDB
+	kill(_h);
+	purge();
```

Comment:
purge makes this O(n) operation. Is it really required here? Wouldn't it be better to call it at the end of commit?